src/scripts/main.js

```diff
@@ -1,3 +1,254 @@
+            table.children[y].children[x].textContent = `${cell}`;
+            table.children[y].children[x].classList.add(`field-cell--${cell}`);
+          }
+        }
+      }
+    }
+
+    move({ moveDirectionCallback, xIsWithinBoard, yIsWithinBoard,
+      nextIndexY, columnStartIndex, nextIndexX,
+      rowStartIndex, currentX, currentY }) {
+      if (xIsWithinBoard) {
+        if (yIsWithinBoard) {
+          const temp = this.board[currentY][currentX];
+          const isTheSameNumber = this.board[nextIndexY][nextIndexX] === temp;
+
+          if ((temp !== 0 && this.board[nextIndexY][nextIndexX] === 0)
+            || isTheSameNumber) {
+            this.board[nextIndexY][nextIndexX] = isTheSameNumber
+              ? temp * 2 : temp;
+            this.board[currentY][currentX] = 0;
+
+            if (isTheSameNumber) {
+              this.updateScore(temp * 2);
+            }
+          }
+
+          return moveDirectionCallback.call(this, nextIndexX, nextIndexY);
+        } else {
+          return moveDirectionCallback.call(
+            this, rowStartIndex, columnStartIndex);
+        }
+      }
+
+      this.appendOneBox();
+      this.occupiedPlaces();
+      this.check2048();
+    }
+
+    moveRight(x = 0, y = 0) {
+      this.move({
+        moveDirectionCallback: this.moveRight,
+        xIsWithinBoard: (y < this.board.length),
+        yIsWithinBoard: (x < this.board.length - 1),
+        nextIndexY: y,
+        columnStartIndex: y + 1,
+        nextIndexX: x + 1,
+        rowStartIndex: 0,
+        currentX: x,
+        currentY: y,
+      });
+    }
+
+    moveLeft(x = this.board.length - 1, y = 0) {
+      this.move({
+        moveDirectionCallback: this.moveLeft,
+        xIsWithinBoard: (y < this.board.length),
+        yIsWithinBoard: (x > 0),
+        nextIndexY: y,
+        columnStartIndex: y + 1,
+        nextIndexX: x - 1,
+        rowStartIndex: (this.board.length - 1),
+        currentX: x,
+        currentY: y,
+      });
+    }
+
+    moveUp(x = 0, y = this.board.length - 1) {
+      this.move({
+        moveDirectionCallback: this.moveUp,
+        xIsWithinBoard: (x < this.board.length),
+        yIsWithinBoard: (y > 0),
+        nextIndexY: y - 1,
+        columnStartIndex: (this.board.length - 1),
+        nextIndexX: x,
+        rowStartIndex: x + 1,
+        currentX: x,
+        currentY: y,
+      });
+    }
+
+    moveDown(x = 0, y = 0) {
+      this.move({
+        moveDirectionCallback: this.moveDown,
+        xIsWithinBoard: (x < this.board.length),
+        yIsWithinBoard: (y < this.board.length - 1),
+        nextIndexY: y + 1,
+        columnStartIndex: 0,
+        nextIndexX: x,
+        rowStartIndex: x + 1,
+        currentX: x,
+        currentY: y,
+      });
+    }
+  }
+
+  const htmlBoard = new Board();
+
+  document.addEventListener('keydown', ev => {
+    if (ev.key === 'ArrowLeft') {
+      htmlBoard.moveLeft();
+    }
+
+    if (ev.key === 'ArrowRight') {
```

Comment:
Will be better if you create a variables for these words and use it here

Reply:
i don't understand this suggestion and thin one below